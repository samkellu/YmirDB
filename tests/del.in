set a 123 33 44
set b 12 a 55
set c 800 12 33
list entries
del c
del d
list entries
list keys
get c
del a
list entries
list keys
get a
bye
