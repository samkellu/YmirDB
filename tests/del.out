==713== Memcheck, a memory error detector
==713== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==713== Using Valgrind-3.15.0 and LibVEX; rerun with -h for copyright info
==713== Command: ./ymirdb
==713== 
==713== error calling PR_SET_PTRACER, vgdb might block
> ok

> ok

> ok

> c [800 12 33]
b [12 a 55]
a [123 33 44]

==713== Invalid read of size 1
==713==    at 0x483EF46: strlen (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
==713==    by 0x48CCD44: __vfprintf_internal (vfprintf-internal.c:1688)
==713==    by 0x48B5D6E: printf (printf.c:33)
==713==    by 0x109799: command_list_entries (ymirdb.c:85)
==713==    by 0x10CD60: main (ymirdb.c:811)
==713==  Address 0x4a49f40 is 0 bytes inside a block of size 216 free'd
==713==    at 0x483DFAF: realloc (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
==713==    by 0x109E91: command_del (ymirdb.c:171)
==713==    by 0x10CE15: main (ymirdb.c:822)
==713==  Block was alloc'd at
==713==    at 0x483DFAF: realloc (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
==713==    by 0x10A0BF: command_set (ymirdb.c:211)
==713==    by 0x10CE98: main (ymirdb.c:826)
==713== 
==713== Invalid read of size 1
==713==    at 0x483EF54: strlen (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
==713==    by 0x48CCD44: __vfprintf_internal (vfprintf-internal.c:1688)
==713==    by 0x48B5D6E: printf (printf.c:33)
==713==    by 0x109799: command_list_entries (ymirdb.c:85)
==713==    by 0x10CD60: main (ymirdb.c:811)
==713==  Address 0x4a49f41 is 1 bytes inside a block of size 216 free'd
==713==    at 0x483DFAF: realloc (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
==713==    by 0x109E91: command_del (ymirdb.c:171)
==713==    by 0x10CE15: main (ymirdb.c:822)
==713==  Block was alloc'd at
==713==    at 0x483DFAF: realloc (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
==713==    by 0x10A0BF: command_set (ymirdb.c:211)
==713==    by 0x10CE98: main (ymirdb.c:826)
==713== 
==713== Invalid read of size 1
==713==    at 0x48436E0: mempcpy (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
==713==    by 0x48E3661: _IO_new_file_xsputn (fileops.c:1236)
==713==    by 0x48E3661: _IO_file_xsputn@@GLIBC_2.2.5 (fileops.c:1197)
==713==    by 0x48CB12B: __vfprintf_internal (vfprintf-internal.c:1688)
==713==    by 0x48B5D6E: printf (printf.c:33)
==713==    by 0x109799: command_list_entries (ymirdb.c:85)
==713==    by 0x10CD60: main (ymirdb.c:811)
==713==  Address 0x4a49f40 is 0 bytes inside a block of size 216 free'd
==713==    at 0x483DFAF: realloc (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
==713==    by 0x109E91: command_del (ymirdb.c:171)
==713==    by 0x10CE15: main (ymirdb.c:822)
==713==  Block was alloc'd at
==713==    at 0x483DFAF: realloc (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
==713==    by 0x10A0BF: command_set (ymirdb.c:211)
==713==    by 0x10CE98: main (ymirdb.c:826)
==713== 
> ok

> b [12 a 55]
a [123 33 44]

> b
a

> no such key

> not permitted

> b [12 a 55]
a [123 33 44]

> b
a

> [123 33 44]

> bye
==713== 
==713== HEAP SUMMARY:
==713==     in use at exit: 0 bytes in 0 blocks
==713==   total heap usage: 59 allocs, 59 frees, 2,744 bytes allocated
==713== 
==713== All heap blocks were freed -- no leaks are possible
==713== 
==713== For lists of detected and suppressed errors, rerun with: -s
==713== ERROR SUMMARY: 6 errors from 3 contexts (suppressed: 0 from 0)
